operations-stdlib/src/commonMain/kotlin/string/Trim.kt

```diff
@@ -0,0 +1,61 @@
+package string
+
+import operation.StandardLogicOperation
+import utils.asList
+import utils.secondOrNull
+import utils.thirdOrNull
+
+object Trim : StandardLogicOperation, StringUnwrapStrategy {
+    override fun evaluateLogic(expression: Any?, data: Any?): Any?  {
+        return when (val firstItem = expression.asList.firstOrNull()) {
+            is String -> firstItem.trimElements(' ', TrimMode.BothEnds)
+            is List<*> -> {
+                val text = firstItem.firstOrNull()
+                val sign = firstItem.secondOrNull() ?: " "
+                val mode = (firstItem.thirdOrNull() as? String).toTrimMode() ?: TrimMode.BothEnds
+                (sign as? String)?.let {
+                    val char: Char = it.single()
+                    text.trimElements(char, mode)
+                }
+            }
+            else -> null
+        }
+    }
+
+    private fun String?.toTrimMode() = when (this) {
+        "start" -> TrimMode.Start
+        "end" -> TrimMode.End
+        "bothEnds" -> TrimMode.BothEnds
+        else -> null
+    }
+
+    private fun Any?.trimElements(char: Char, mode: TrimMode) =
+        when (this) {
+            is String -> {
+                modeBasedTrim(
+                    mode = mode,
+                    start = { trimStart(char) },
+                    end = { trimEnd(char) },
+                    bothEnds = { trim(char) }
+                )
+            }
+            else -> null
+        }
+
+    private fun modeBasedTrim(mode: TrimMode,
```

Comment:
actually.. why make this so complicated? With highe order functions, providing from other place and not just

```
    private fun Any?.trimElements(char: Char, mode: TrimMode) =
        when (this) {
            is String -> mode.trimByMode(this, char)
            else -> null
        }
```

```
    private fun TrimMode.trimByMode(target: String, char: Char) =
        when (this) {
            TrimMode.Start -> target.trimStart(char)
            TrimMode.End -> target.trimEnd(char)
            TrimMode.BothEnds -> target.trim(char)
        }
```

less 'any' usage ;)

Reply:
👍 I changed it as @KTGypsy suggested